core/src/com/unciv/logic/map/mapunit/movement/UnitMovement.kt

```diff
@@ -807,4 +854,13 @@ class PathsToTilesWithinTurn : LinkedHashMap<Tile, UnitMovement.ParentTileAndTot
         }
         return reversePathList.reversed()
     }
+    
+    fun removeAllTilesNotInSet(otherPathsToTiles: PathsToTilesWithinTurn) {
+        val iterator = this.iterator()
+        while(iterator.hasNext()) {
+            if (!otherPathsToTiles.contains(iterator.next().key)) {
+                iterator.remove()
+            }
+        }
```

Comment:
how about .removeAll {  }, we don't need a special function for this

Reply:
I used .removeIf {} and removed that method entirely since it had only one use.